plugins/decklink/decklink-device-instance.cpp

```diff
@@ -83,15 +95,19 @@ void DeckLinkDeviceInstance::HandleAudioPacket(
 	currentPacket.frames      = frameCount;
 	currentPacket.timestamp   = timestamp;
 
-	if (!ISSTEREO(channelFormat)) {
+	int maxdevicechannel = device->GetMaxChannel();
+
+	if ((channelFormat != SPEAKERS_UNKNOWN)&&(channelFormat != SPEAKERS_MONO)&&
+		(channelFormat != SPEAKERS_STEREO)&&(channelFormat != SPEAKERS_2POINT1)
+		&& (maxdevicechannel >= 8) ) {
 		if (audioRepacker->repack((uint8_t *)bytes, frameCount) < 0) {
 			LOG(LOG_ERROR, "Failed to convert audio packet data");
 			return;
 		}
-
-		currentPacket.data[0]   = (*audioRepacker)->packet_buffer;
-	} else {
-		currentPacket.data[0]   = (uint8_t *)bytes;
+		currentPacket.data[0] = (*audioRepacker)->packet_buffer;
+	}
+	else {
```

Comment:
Style:  This `else` should be on the same line as the preceding closing curly brace.